### PR TITLE
Proofpoint 1.15 changes

### PR DESCRIPTION
**Pull request by snelson-pp**
Porting changes from our kops 1.15 branch

### CODE REVIEW

**Comment by johngmyers** [comment]
Tests are failing. Each of those commits "update-expected.sh" needs to be edited, running `hack/update-expected.sh` to regenerate the expected values.

**Comment by johngmyers** [comment]
What happened to f7501d5fd3037a3a1a53f64a1569b103b914e6eb ?

**Comment by johngmyers** [comment]
We'll need to change `kops-controller` to run as non-root, though that can be a separate PR.

**Comment by snelson-pp** [comment]
> Tests are failing. Each of those commits "update-expected.sh" needs to be edited, running `hack/update-expected.sh` to regenerate the expected values.

Run hack/update-expected.sh locally?

**Comment by johngmyers** [comment]
Yes.

**Comment by johngmyers** [comment]
We could possibly edit out the previous `update-expected.sh` commits and add a new one at the end of this PR.

File: upup/models/cloudup/resources/addons/dns-controller.addons.k8s.io/k8s-1.12.yaml.template [hunk 1]
```diff
@@ -57,6 +59,17 @@ spec:
...
+          runAsUser: 1000
+      dnsPolicy: Default
+      hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      priorityClassName: system-cluster-critical
+      serviceAccount: dns-controller
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
```
Comment (johngmyers) [inline upup/models/cloudup/resources/addons/dns-controller.addons.k8s.io/k8s-1.12.yaml.template]: A lot of this is duplicated from above. I think you might need to retry the cherrypick.

File: upup/models/cloudup/resources/addons/dns-controller.addons.k8s.io/k8s-1.6.yaml.template [hunk 2]
```diff
@@ -57,7 +58,8 @@ spec:
           requests:
             cpu: 50m
             memory: 50Mi
-
+        securityContext:
+          runAsUser: 1000
```
Comment (johngmyers) [inline upup/models/cloudup/resources/addons/dns-controller.addons.k8s.io/k8s-1.6.yaml.template]: We don't need to change the pre-1.12 manifest as we're not using it.

Comment (snelson-pp) [reply upup/models/cloudup/resources/addons/dns-controller.addons.k8s.io/k8s-1.6.yaml.template]: That was a leftover from previous, but I'll remove it.

Comment (snelson-pp) [reply upup/models/cloudup/resources/addons/dns-controller.addons.k8s.io/k8s-1.6.yaml.template]: Removed and reran update-expected

**Pull request merged**